### ISSUE TITLE
Unexpected side effect using numpy trig functions

### ISSUE THREAD

**Issue by josiahslack** [hgrecco/pint#399]
I ran across this yesterday.

I type the following into a python REPL:

`
import numpy as np
import pint

ureg = pint.UnitRegistry()
Q_ = ureg.Quantity
theta = Q_(30, ureg.degrees)
type(theta.m) # returns int
s_theta = np.sin(theta)
type(theta.m) # returns numpy.ndarray
`
Printing theta, using comparisons, etc. doesn't show that anything has changed about it. I ran into trouble when I was saving an angle's magnitude to a JSON file, and json.dump complained that it didn't know how to write 'array(0.0)'. I worked backwards, and eventually isolated the point where the type of the angle's magnitude changed.

**Comment by hgrecco** [comment]
Thanks for reporting. We should not transform scalars to arrays. 
And it is not numpy's fault
```
>>> type(np.sin(.3))
<class 'numpy.float64'>
>>> type(.3)
<class 'float'>
```

**Comment by CD3** [comment]
I ran into this too.

```
import pint
import numpy

ureg = pint.UnitRegistry()

Angle = ureg.Quantity(30,'degree')

Cos = ureg.Quantity(numpy.sqrt(3.)/2.)
print Angle, type(Angle), Angle.magnitude, type(Angle.magnitude)
print Cos, type(Cos), Cos.magnitude, type(Cos.magnitude)
print "cos of {Angle:.1f} is {Cos:.1e}".format(Angle=Angle, Cos=Cos)

Cos = numpy.cos(Angle)
print Angle, type(Angle), Angle.magnitude, type(Angle.magnitude)
print Cos, type(Cos), Cos.magnitude, type(Cos.magnitude)
print "cos of {Angle:.1f} is {Cos:.1e}".format(Angle=Angle, Cos=Cos)
```

This throws an error on the second `format()` call. 

```
30 degree <class 'pint.unit.Quantity'> 30 <type 'int'>
0.866025403784 dimensionless <class 'pint.unit.Quantity'> 0.866025403784 <type 'numpy.float64'>
cos of 30.0 degree is 8.7e-01 dimensionless
30 degree <class 'pint.unit.Quantity'> 30 <type 'numpy.ndarray'>
0.866025403784 dimensionless <class 'pint.unit.Quantity'> 0.866025403784 <type 'numpy.float64'>
Traceback (most recent call last):
  File "./test.py", line 18, in <module>
    print "cos of {Angle:.1f} is {Cos:.1e}".format(Angle=Angle, Cos=Cos)
  File "/usr/lib64/python2.7/site-packages/pint/quantity.py", line 141, in __format__
    format(obj.magnitude, remove_custom_flags(spec)),
ValueError: Unknown format code 'f' for object of type 'unicode'

```
It took me a while to figure out what was going on. The call to `numpy.cos` changes the type of `Angle`. It isn't just trig functions either, `numpy.abs` will also change the type of its argument.

I would expect any quantities that get passed into the numpy functions are left unmodified. Is `pint` modifying the argument intentionally for some reason? If not, could the numpy hooks just make a copy of the arguments to pass to the numpy functions?

**Comment by CD3** [comment]
A workaround is to pass copies of the quantity to the numpy functions using the `copy` module

```
import pint
import numpy
from copy import *

ureg = pint.UnitRegistry()

Angle = ureg.Quantity(30,'degree')

Cos = ureg.Quantity(numpy.sqrt(3.)/2.)
print Angle, type(Angle), Angle.magnitude, type(Angle.magnitude)
print Cos, type(Cos), Cos.magnitude, type(Cos.magnitude)
print "cos of {Angle:.1f} is {Cos:.1e}".format(Angle=Angle, Cos=Cos)

Cos = numpy.cos(copy(Angle)) # pass copy of quantity so it won't be modified.
print Angle, type(Angle), Angle.magnitude, type(Angle.magnitude)
print Cos, type(Cos), Cos.magnitude, type(Cos.magnitude)
print "cos of {Angle:.1f} is {Cos:.1e}".format(Angle=Angle, Cos=Cos)
```
works as expected
```
30 degree <class 'pint.unit.Quantity'> 30 <type 'int'>
0.866025403784 dimensionless <class 'pint.unit.Quantity'> 0.866025403784 <type 'numpy.float64'>
cos of 30.0 degree is 8.7e-01 dimensionless
30 degree <class 'pint.unit.Quantity'> 30 <type 'int'>
0.866025403784 dimensionless <class 'pint.unit.Quantity'> 0.866025403784 <type 'numpy.float64'>
cos of 30.0 degree is 8.7e-01 dimensionless
```

**Comment by hgrecco** [comment]
Review after #905

**Comment by hgrecco** [comment]
Solved by #905
